presto-main/src/main/java/com/facebook/presto/sql/planner/LocalExecutionPlanner.java

```diff
@@ -1139,7 +1139,7 @@ public PhysicalOperation visitProject(ProjectNode node, LocalExecutionPlanContex
                 sourceNode = filterNode.getSource();
                 filterExpression = Optional.of(filterNode.getPredicate());
                 Expression temp = filterNode.getPredicateWithoutTupleDomain();
-                if (!temp.equals(TRUE_LITERAL)) {
+                if (temp != null && !temp.equals(TRUE_LITERAL)) {
```

Comment:
```suggestion
                if (!TRUE_LITERAL.equals(temp)) {
```

Reply:
@tdcmeehan Thanks! The semantics is slightly different. With this change the next line would need to changed to Optional.ofNullable. I'll update it anyways.